File: src/01-easy/power-of-thor-episode-1/solution.cpp
#include <iostream>
#include <string>
using namespace std;
int main() {
    int lx, ly, tx, ty, e;
    cin >> lx >> ly >> tx >> ty;
    while (1) {
        cin >> e;
        string dir;
        if (ty > ly) { dir += "N"; ty--; }
        else if (ty < ly) { dir += "S"; ty++; }
        if (tx > lx) { dir += "W"; tx--; }
        else if (tx < lx) { dir += "E"; tx++; }
        cout << dir << endl;
    }
}

File: src/01-easy/power-of-thor-episode-1/solution.go
package main
import "fmt"
func main() {
    var lx, ly, tx, ty, e int
    fmt.Scan(&lx, &ly, &tx, &ty)
    for {
        fmt.Scan(&e)
        dir := ""
        if ty > ly { dir += "N"; ty-- } else if ty < ly { dir += "S"; ty++ }
        if tx > lx { dir += "W"; tx-- } else if tx < lx { dir += "E"; tx++ }
        fmt.Println(dir)
    }
}
